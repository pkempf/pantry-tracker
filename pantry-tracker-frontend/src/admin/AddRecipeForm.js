import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import Container from "react-bootstrap/Container";
import PantryForm from "../forms/PantryForm";
import UserContext from "../UserContext";
import AlertContext from "../AlertContext";
import PantryApi from "../api";

const AddRecipeForm = () => {
  const user = useContext(UserContext);
  const { setMessage } = useContext(AlertContext);

  if (!user.isAdmin) {
    return <Redirect to="/" />;
  }

  const formFields = [
    {
      name: "name",
      label: "Name",
      inputType: "text",
      initialValue: "",
      placeholder: "Example: Banana pancakes",
      required: true,
    },
    {
      name: "instructions",
      label: "Instructions",
      inputType: "text",
      initialValue: "",
      placeholder: "Enter recipe instructions here",
      required: true,
      asOverride: "textarea",
      styleOverride: { height: "100px" },
    },
    {
      name: "category",
      label: "Category",
      inputType: "text",
      initialValue: "",
      placeholder: "Example: Dessert",
      required: false,
    },
    {
      name: "area",
      label: "Area",
      inputType: "text",
      initialValue: "",
      placeholder: "Example: Italian",
      required: false,
    },
  ];

  async function add(recipe) {
    try {
      let newRecipe = await PantryApi.addRecipe(recipe);
      if (newRecipe.name && newRecipe.name === recipe.name) {
        setMessage({
          text: `Added recipe ${newRecipe.name}!`,
          variant: "success",
        });
      }
    } catch (err) {
      setMessage({
        text: `Error adding recipe ${recipe.name}`,
        variant: "danger",
      });
    }
  }

  return (
    <Container className="AddRecipeForm justify-content-center">
      <PantryForm
        formTitle={`Add Recipe`}
        fields={formFields}
        submitButtonText="Add Recipe"
        processData={(data) => add(data)}
        backButtonTo="/admin"
      />
    </Container>
  );
};

export default AddRecipeForm;
