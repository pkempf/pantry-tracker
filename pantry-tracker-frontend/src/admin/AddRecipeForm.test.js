import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AddRecipeForm from "./AddRecipeForm";
import UserContext from "../UserContext";
import AlertContext from "../AlertContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <AlertContext.Provider value={{ setMessage: {} }}>
          <AddRecipeForm />
        </AlertContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <AlertContext.Provider value={{ setMessage: {} }}>
          <AddRecipeForm />
        </AlertContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
