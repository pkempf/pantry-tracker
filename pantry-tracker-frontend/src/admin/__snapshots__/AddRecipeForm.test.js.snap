// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddRecipeForm justify-content-center container"
  >
    <div
      class="PantryForm justify-content-center row"
    >
      <div
        class="col-md-8 col-sm-10"
      >
        <div
          class="card"
        >
          <h4
            class="card-header"
          >
            Add Recipe
          </h4>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="form-control"
                  id="name"
                  name="name"
                  placeholder="Example: Banana pancakes"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="instructions"
                >
                  Instructions
                </label>
                <textarea
                  class="form-control"
                  id="instructions"
                  name="instructions"
                  placeholder="Enter recipe instructions here"
                  required=""
                  style="height: 100px;"
                  type="text"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="category"
                >
                  Category
                </label>
                <input
                  class="form-control"
                  id="category"
                  name="category"
                  placeholder="Example: Dessert"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="area"
                >
                  Area
                </label>
                <input
                  class="form-control"
                  id="area"
                  name="area"
                  placeholder="Example: Italian"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="mr-2 btn btn-primary"
                type="submit"
              >
                Add Recipe
              </button>
              <a
                class="btn btn-secondary"
                href="/admin"
              >
                Back
              </a>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
